### PR TITLE
Refactoring: converted to TypeScript

### PR DESCRIPTION
**Pull request by rubenthoms**
Adjusted build environment and converted all files to TypeScript except for the main component files (changed those to JSX).

### CODE REVIEW

File: src/lib/components/Select/Select.jsx [hunk 1]
```diff
@@ -8,48 +8,50 @@
 import React, { Component } from "react";
 import PropTypes from "prop-types";
 import "./Select.css";
+
 /**
  * Select is a dash wrapper for the html select tag.
  */
-export default class Select extends Component {
-    handleChange(e) {
+const Select = ({ id, size, options, value, multi, className, style, parent_className, parent_style, setProps, persistence, persisted_props, persistence_type }) => {
```
Comment (HansKallekleiv) [inline src/lib/components/Select/Select.jsx:15]: too long line?

File: src/lib/components/WebvizPluginPlaceholder/WebvizPluginPlaceholder.jsx [hunk 2]
```diff
@@ -30,154 +29,153 @@ import "./webviz_plugin_component.css";
...
-        this.state = {
-            expanded: false,
-            showOverlay: false,
-            tourIsOpen: false,
-        };
-    }
-
-    componentDidUpdate(_, prevState) {
-        if (this.props.download !== null && this.props.download !== "") {
+const WebvizPluginPlaceholder = ({ id, children, buttons, contact_person, download, screenshot_filename, tour_steps, data_requested, setProps }) => {
```
Comment (HansKallekleiv) [inline src/lib/components/WebvizPluginPlaceholder/WebvizPluginPlaceholder.jsx:32]: too long line?

**Review by HansKallekleiv — CHANGES_REQUESTED** [review]

**Review by HansKallekleiv — APPROVED** [review]
LGTM 👍

File: .babelrc [hunk 3]
```diff
@@ -1,9 +1,16 @@
...
+        }
+      }
+    ],
     "@babel/preset-react"
   ],
   "plugins": [
     "@babel/plugin-proposal-class-properties"
   ]
-}
+}
```
Comment (asnyv) [inline .babelrc:16]: newline

**Pull request merged**